airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncherUTest.kt

```diff
@@ -0,0 +1,135 @@
+            catalog,
+            syncManager,
+            inputConsumerTask,
+            spillToDiskTaskFactory,
+            setupTaskFactory,
+            openStreamTaskFactory,
+            processRecordsTaskFactory,
+            processFileTaskFactory,
+            processBatchTaskFactory,
+            closeStreamTaskFactory,
+            teardownTaskFactory,
+            flushCheckpointsTaskFactory,
+            timedFlushTask,
+            updateCheckpointsTask,
+            exceptionHandler,
+            useFileTranfer
+        )
+    }
+
+    // private val mainThread = newSingleThreadContext("UI thread")
+
+    @Test
+    fun `test that we don't start the spill-to-disk task when file transfer is enabled`() =
+        runTest {
+            coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+            coEvery { taskScopeProvider.launch(any()) } returns Unit
+
+            val destinationTaskLauncher = getDefaultDestinationTaskLauncher(true)
+            // This is needed to let the run method to complete
+            destinationTaskLauncher.handleTeardownComplete()
+            destinationTaskLauncher.run()
+
+            coVerify { catalog wasNot Called }
+            coVerify { spillToDiskTaskFactory wasNot Called }
+        }
+
+    @Test
+    fun `test that we start the spill-to-disk task when file transfer is disabled`() = runTest {
+        coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+        coEvery { taskScopeProvider.launch(any()) } returns Unit
+
+        val stream = mockk<DestinationStream>(relaxed = true)
+        val streamDescriptor = DestinationStream.Descriptor("namespace", "name")
+        every { stream.descriptor } returns streamDescriptor
+        coEvery { catalog.streams } returns listOf(stream)
+
+        val spillToDiskTask = mockk<SpillToDiskTask>(relaxed = true)
+        coEvery { spillToDiskTaskFactory.make(any(), any()) } returns spillToDiskTask
+
+        val destinationTaskLauncher = getDefaultDestinationTaskLauncher(false)
+        // This is needed to let the run method to complete
+        destinationTaskLauncher.handleTeardownComplete()
+        destinationTaskLauncher.run()
+
+        coVerify { spillToDiskTaskFactory.make(any(), any()) }
+        coVerify { exceptionHandler.withExceptionHandling(spillToDiskTask) }
+    }
+
+    @Test
+    fun `test handle file`() = runTest {
+        coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+        coEvery { taskScopeProvider.launch(any()) } returns Unit
+        val processFileTask = mockk<ProcessFileTask>(relaxed = true)
+        every { processFileTaskFactory.make(any(), any(), any()) } returns processFileTask
+
+        val destinationTaskLauncher = getDefaultDestinationTaskLauncher(true)
+        destinationTaskLauncher.handleFile(mockk(), mockk())
+
+        coVerify { exceptionHandler.withExceptionHandling(processFileTask) }
```

Comment:
Technically the contract here is that the task returned by the exception handler gets passed to TaskScopeProvider.launch. But that's tested elsewhere, and also it would be pretty unwieldy to try to implement here with a capture.

Reply:
I have updated the test to check that the launch command is properly call. I am not sure where the capture will help, it feels to me that it is checking that the mockk library returns the right mock